### DISCUSSION TITLE
Visual code studio not working with training

### DISCUSSION THREAD

**Discussion by creativename00** [ultralytics/ultralytics#15021] [Q&A]
I'm able to run my model, but when attempting to train it gets 90% of the way there on the first epoch and hangs up. If I run the same code in python by itself, then it works. Main difference I see is the python version is using dataloaders, while the vs code version is using 0. Tried updating or reverting the python extension version with no luck. Any ideas?

**Comment by pderrenger** [comment]
@creativename00 it sounds like the issue might be related to how Visual Studio Code handles the environment or resources. First, ensure you are using the latest version of the Ultralytics package. If the issue persists, try running VS Code as an administrator or adjusting the integrated terminal settings. Additionally, check if there are any conflicts with other extensions or settings in VS Code that might affect the DataLoader. If the problem continues, running the training script directly in a standalone Python environment might be the best approach.

**Reply by creativename00** [reply]
> @creativename00 it sounds like the issue might be related to how Visual Studio Code handles the environment or resources. First, ensure you are using the latest version of the Ultralytics package. If the issue persists, try running VS Code as an administrator or adjusting the integrated terminal settings. Additionally, check if there are any conflicts with other extensions or settings in VS Code that might affect the DataLoader. If the problem continues, running the training script directly in a standalone Python environment might be the best approach.

I believe I have discovered the issue. If I roll back vs codes python version to the same one I had in the standalone environment it works. The version that works is 3.11.8. If I try using a newer standalone python it does the same thing and hangs up.

**Reply by pderrenger** [reply]
@creativename00 thank you for the update. It appears the issue is related to the Python version compatibility. To ensure smooth operation, you might want to stick with Python 3.11.8 for now. Additionally, please make sure you are using the latest version of the Ultralytics package, as updates often include important fixes and improvements. If you encounter any further issues, feel free to reach out here.
